src/core/content/util.ts

```diff
@@ -747,7 +751,14 @@ export function injectScriptIntoDocument(scriptUrl: string): void {
  * @param logType - Log type
  */
 /* istanbul ignore next */
-export function debugLog(text: unknown, logType: DebugLogType = 'log'): void {
+export async function debugLog(text: unknown, logType: DebugLogType = 'log'): Promise<void> {
```

Comment:
This should not be async.
By being async, the order of debug logs will not be the same as the order of the extension calling debugLog, which has the potential to cause confusion when debugging.

Make the function synchronous, and instead of doing the check inside the function, I think I'd suggest creating a queue class, something like (and im typing this on my phone so this probably does not even compile much less be good code, but you get the point):
```
class Queue {
  private queue: [any, DebugLogType][] = [];
  private isActive = true;
  private shouldPrint = Options.getOption(Options.DEBUG_LOGGING_ENABLED);
  
  private async start() {
    if (this.isActive) return;
    this.isActive = false;
    for (const i = 0; i < 100 && this.queue.length > 0; i++) { // i imagine some fixed upper limit is prolly good idea
      const cur = this.queue.shift();
      if (await this.shouldPrint) {
        // print
      }
    }
  }
  
  public push(text: unknown, logType: DebugLogType) {
    queue.push([text, logType]);
    this.start();
  }
}
const queue = new Queue();

export function debugLog(text:unknown, logType....
  queue.push(text, logType);
```
This approach also helps reduce IO access a bit, not that I expect that to be a big issue anyway though, but I don't see a solid reason not to.

Reply:
The queue should of course also get a more descriptive name

Reply:
@yayuyokitano I've added a queue for log messages now, based on your suggested implementation: 83336ce